awx/api/generics.py

```diff
@@ -859,15 +849,15 @@ def _get_copy_return_serializer(self, *args, **kwargs):
     def _decrypt_model_field_if_needed(obj, field_name, field_val):
         if field_name in getattr(type(obj), 'REENCRYPTION_BLACKLIST_AT_COPY', []):
             return field_val
-        if isinstance(field_val, dict):
+        if isinstance(obj, Credential) and field_name == 'inputs':
+            for secret in obj.credential_type.secret_fields:
+                if secret in field_val:
+                    field_val[secret] = decrypt_field(obj, secret)
+        elif isinstance(field_val, dict):
             for sub_field in field_val:
                 if isinstance(sub_field, str) \
                         and isinstance(field_val[sub_field], str):
-                    try:
-                        field_val[sub_field] = decrypt_field(obj, field_name, sub_field)
-                    except AttributeError:
-                        # Catching the corner case with v1 credential fields
-                        field_val[sub_field] = decrypt_field(obj, sub_field)
+                    field_val[sub_field] = decrypt_field(obj, field_name, sub_field)
```

Comment:
there are some 🐉 s here. Was there really a different means of encryption for credentials created from `/api/v1/credentials/`? If so, what's the behavior of those old credentials that came over from the migration? The current level of testing might not have captured those. Since there's no v1 / v2 distinction of the models, the users can't tell the difference between credentials created from one or the other.

Reply:
So up until now, we're relied on `decrypt_field` using `getattr(cred, 'field')`.  In this PR, I've removed the special `__getattr__` that facilitates this - you *always* have to specify `cred.inputs[...]`.

 I can't recall the details now (I wrote this in ~March), but I ended up having to do this as a way to make this work now that `__getattr__` is gone on `Credential`.